libs/dyn/convert/from_typed.go

```diff
@@ -142,18 +164,17 @@ func fromTypedString(src reflect.Value, ref dyn.Value) (dyn.Value, error) {
 
 		return dyn.V(value), nil
 	case dyn.KindNil:
-		// This field is not set in the reference, so we only include it if it has a non-zero value.
-		// Otherwise, we would always include all zero valued fields.
-		if src.IsZero() {
+		if src.IsZero() && !slices.Contains(options, includeZeroValues) {
 			return dyn.NilValue, nil
 		}
+		// This field is not set in the reference, so we only include it if it
+		// has a non-zero value or the includeZeroValues option is set.
```

Comment:
The comments seem unmatched to the if clause now.

Reply:
done